test/jdk/java/lang/ClassLoader/loadLibraryDeadlock/TestLoadLibraryDeadlock.java

```diff
@@ -84,12 +88,15 @@ private static OutputAnalyzer genKey() throws Throwable {
         );
     }
 
-    private static OutputAnalyzer createJar(String outputJar, String... classes) throws Throwable {
-        String jar = JDKToolFinder.getJDKTool("jar");
-        List<String> commands = new ArrayList<>();
-        Collections.addAll(commands, jar, "cvf", outputJar);
-        Collections.addAll(commands, classes);
-        return runCommandInTestClassPath(commands.toArray(new String[0]));
+    private static void createJar(String outputJar, String... classes) throws Throwable {
+        List<String> args = new ArrayList<>();
+        Collections.addAll(args, "cvf", Paths.get(testClassPath, outputJar).toString());
+        for (String c : classes) {
+            Collections.addAll(args, "-C", testClassPath, c);
```

Comment:
I believe only one single `-C` option is good enough for this case since the classes are all in one single directory.

Reply:
Contrary to the man which says "-C DIR Changes the specified directory and includes the files specified at the end of the command line.", here is how it works:

```
bash-3.2$ find .
.
./dir
./dir/B.class
./dir/A.class
bash-3.2$ jar cvf a.jar -C dir A.class B.class
B.class : no such file or directory
added manifest
adding: A.class(in = 176) (out= 149)(deflated 15%)
```

and no jar is produced. Lines 614 and 647 of src/jdk.jartool/share/classes/sun/tools/jar/Main.java explain why. IIRC the intent was to mimic tar -C behaviour where -C applies to all files that follow, but jar does not work that way currently. 

It's probably worth reaching consensus that it's a problem worth fixing first (a scary one to fix by the way, provided how many tools could rely on the current behaviour). Alternatively, man could be adjusted to reflect the current order. I could work on that as a follow up.

Reply:
You are probably right.  I have no more comment.